File: CSharp/Solutions/PalindromeNumber.cs
namespace Solutions;

public class PalindromeNumber {
    public static bool IsPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        static int reverse(int n) {
            int result = 0;
            while (n != 0) {
                int lastDigit = n % 10;
                result = (result * 10) + lastDigit;
                n /= 10;
            }
            return result;
        }

        return reverse(x) == x;
    }
}

File: Go/palindrome_number/palindrome_number.go
package palindrome_number

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}

	temp := x
	reversed := 0

	for temp != 0 {
		lastDigit := temp % 10
		reversed = reversed*10 + lastDigit
		temp /= 10
	}

	return reversed == x
}
